proxy/src/main/java/com/velocitypowered/proxy/connection/client/LoginSessionHandler.java

```diff
@@ -82,6 +82,7 @@ public class LoginSessionHandler implements MinecraftSessionHandler {
   private @MonotonicNonNull ServerLogin login;
   private byte[] verify = EMPTY_BYTE_ARRAY;
   private @MonotonicNonNull ConnectedPlayer connectedPlayer;
+  private boolean receivedEncryptionResponse = false;
```

Comment:
While this works I think this class should have a state enum for tracking the process rather. Just a note for future refactors

Reply:
This was honestly my thought too, just figured I'd leave this PR to those more involved with this